sql/plan/bindvar.go

```diff
@@ -26,19 +26,20 @@ import (
 // returned and the |BindVar| expression is left in place. There is no check on
 // whether all entries in |bindings| are used at least once throughout the |n|.
 // sql.DeferredType instances will be resolved by the binding types.
-func ApplyBindings(n sql.Node, bindings map[string]sql.Expression) (sql.Node, error) {
-	n, _, err := applyBindingsHelper(n, bindings)
+func ApplyBindings(n sql.Node, bindings map[string]sql.Expression, usedBindings map[string]bool) (sql.Node, error) {
```

Comment:
Better to return this as a new result param than to require it to be passed in to be modified here